tools/bump-version.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set -eu -o pipefail
+shopt -s lastpipe
+
+FC_TOOLS_DIR=$(dirname $(realpath $0))
+source "$FC_TOOLS_DIR/functions"
+FC_ROOT_DIR=$FC_TOOLS_DIR/..
+
+
+if [ $# -ne 1 ]; then
+    cat <<EOF
+$0 <version>
+
+    Example: $0 1.4.0-dev
+
+    Bump Firecracker release version:
+    1. Updates Cargo.toml / Cargo.lock
+    2. Runs `cargo update`
+EOF
+    exit 1
+fi
+version=$1
+
+
+function check_snapshot_version {
+    local version=$1
+    local snap_version=$(echo $version |cut -f-2 -d. |tr . _)
+    if ! grep -s FC_V${snap_version}_SNAP_VERSION src/vmm/src/version_map.rs; then
+       die "I couldn't find FC_V${snap_version}_SNAP_VERSION in src/vmm/src/version_map.rs"
+    fi
+}
+
+check_snapshot_version "$version"
+
+
+# Get current version from the swagger spec.
+prev_ver=$(get_swagger_version)
+
+say "Updating from $prev_ver to $version ..."
+# Update version in files.
+files_to_change=(
+    "$FC_ROOT_DIR/src/api_server/swagger/firecracker.yaml"
+    "$FC_ROOT_DIR/src/firecracker/Cargo.toml"
+    "$FC_ROOT_DIR/src/jailer/Cargo.toml"
+    "$FC_ROOT_DIR/src/rebase-snap/Cargo.toml"
+    "$FC_ROOT_DIR/src/seccompiler/Cargo.toml"
+)
+say "Updating source files:"
+for file in "${files_to_change[@]}"; do
+    say "- $file"
+    # Dirty hack to make this work on both macOS/BSD and Linux.
+    # FIXME This is very hacky and can unintentionally bump other versions, so
+    # only do the replacement *once*.
```

Comment:
Since this could lead to unintentional behaviors, I think we should create an issue for it and use "TODO" with the link.
https://github.com/firecracker-microvm/firecracker/blob/main/.github/pull_request_template.md?plain=1#L24

Reply:
Actually the comment is old. The new replacements are much safer.

Reply:
But that is still the case, right?
e.g. Cargo.toml
```
[[package]]
version = "1.3.0"

[dependencies]
hoge = { version = "1.3.0", features = "fuga" }
```

Reply:
How about adding "if" condition for running separate `sed` command for YAML and TOML?
If doing so, it can be more resistant against the above issue by adding `^` in the regex pattern.
```
    # For TOML
    sed -i "s/^version = \"$prev_ver\"/version = \"$version\"/" "$file"
```